src/murmur/Messages.cpp

```diff
@@ -601,6 +602,13 @@ void Server::msgAuthenticate(ServerUser *uSource, MumbleProto::Authenticate &msg
 		sendMessage(uSource, mptm);
 	}
 
+	if (msg.m_client_type() == static_cast<int>(ClientType::BOT)) {
+		uSource->m_clientType = ClientType::BOT;
+		m_botCount++;
+	} else {
+		uSource->m_clientType = ClientType::REGULAR;
+	}
```

Comment:
```suggestion
	switch (msg.client_type()) {
		case static_cast< int >(ClientType::BOT):
			uSource->m_clientType = ClientType::BOT;
			m_botCount++;
			break;
		case static_cast< int >(ClientType::REGULAR):
			// No-op (also applies to unknown values of msg.client_type())
			// (The default client type is regular anyway, so we don't need to change anything here)
			break;
	}
```
This has the advantage that the compiler will generate a warning for new enum entries that might get added in the future that are not yet handled here. In order to actually get such a warning, we must not use a `default` clause in our switch. However, since we want a no-op for unknown types anyway, that fits nicely.

Reply:
Thank you for a review, very informative. I implemented your suggestions locally before pushing it and tested if everything worked just to be sure